### ISSUE TITLE
Allow abort glob progress

### ISSUE THREAD

**Issue by otakustay** [sindresorhus/globby#261]
I was racing a globby call with a 200ms timeout so that I can fallback to a default value if glob takes too long, I’d also like to abort glob progress if timeout happens to save some filesystem scan and cpu usage, how about if we can pass an `AbortSignal` in oprions?

Another case is to implement something like `globExists` which returns true on first match, this is done by using `globStream` function and directly return true in the `for … await` loop, also I’d like to abort further progress of glob

Of course it’s a best if a `globExists` function can be provided.

**Comment by sindresorhus** [comment]
Makes sense, but it will have to be implemented in https://github.com/mrmlnc/fast-glob first. Open an issue there too.

**Comment by sindresorhus** [comment]
This still needs fast-glob 3.4.0/4.0.0 to be released. When it is, it should *just work* here.
